public/app/plugins/datasource/elasticsearch/query_builder.js

```diff
@@ -243,19 +242,26 @@ function (queryDef) {
         }
       });
     }
-
+    var size = 500;
+    if(queryDef.size){
+      if(this.esVersion >= 5 && queryDef.size !== 0) {
```

Comment:
do we needs this? Cant we just check if size is defined on the query then use that, if it's not then use the default.